### DISCUSSION TITLE
NumberInput for percentage value results in bad UX

### DISCUSSION THREAD

**Discussion by Meierschlumpf** [mantinedev/mantine#1573] [Q&A] [answered]
When using the NumberInput with its formatter and parser I can add a % sign at the end. 
However the cursor moves directly to the end, instead of to the front of the % sign. 
Would be easy to fix, if a listener would be available for running a function after the formatter has edited the input field.
Is there any build in or third party solution for that?

My NumberInput:
```tsx
<NumberInput
   name='percentageValue'
   label='PercentageValue'
   required
   hideControls
   parser={(value: string | undefined) => value?.replace(/%?/g, "")}
   formatter={(value) =>
      value && !Number.isNaN(parseFloat(value)) ? `${value}%` : ""
   }
/>
```
Results in 
![image](https://user-images.githubusercontent.com/63781622/172046623-dcb2fb69-4c8f-4dd4-b071-d7fca5d6d580.png)

**Comment by rtivital** [comment, marked as answer]
I doubt that we should manage cursor position on library side since there are countless possibilities for number formatting. I think you can manipulate cursor position with input ref.

**Reply by Meierschlumpf** [reply]
yeah thats true you can edit that, but only before the formating happens. And after the formating takes place the cursor is placed again on the back. (I tried it with onChange).
